elftools/elf/elffile.py

```diff
@@ -549,7 +549,11 @@ def get_shstrndx(self):
         if self['e_shstrndx'] != SHN_INDICES.SHN_XINDEX:
             return self['e_shstrndx']
         else:
-            return self._get_section_header(0)['sh_link']
+            header = self._get_section_header(0)
```

Comment:
I'm not sure why this protection exists. Here we can't even get the section header for the first section in the file -- what kind of corruption is this and what do you expect to happen?

Reply:
This protection comes from a malware sample I tried to open with `ELFFile` but the `__init__` crashed trying to access the tampered section header.

[This](https://www.virustotal.com/gui/file/8b9bfe8d5d32d7218059fcd24660a15a177a4ee75670cc1af86b357d59903cc7/detection) is one of the multiple samples that crash when I try to open it. This sample is completely executable although their headers are a little bit tampered as you can see in VirusTotal in the ["Details" tab](https://www.virustotal.com/gui/file/8b9bfe8d5d32d7218059fcd24660a15a177a4ee75670cc1af86b357d59903cc7/details), "ELF info" section.

I expect `ELFFile` doesn't crash during the initialization as I need to get info from the program headers. I'm using `pyelftools` [here](https://github.com/NozomiNetworks/upx-recovery-tool/blob/main/upx-recovery-tool.py#L187) to know the end of the program headers. The file ([test/testfiles_for_unittests/corrupt_e_shstrndx.elf](https://github.com/eliben/pyelftools/pull/440/files#diff-22865a7128b02800febcf128350860b88afb51c44b823d6bd28bc0223c6bf0a9)) has similar headers to that malware samples.

Reply:
This protection exists to fool or crash some security tools [like mine](https://github.com/NozomiNetworks/upx-recovery-tool/blob/main/upx-recovery-tool.py#L187).